core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -363,6 +363,10 @@ private ManifestFile filterManifest(
   }
 
   private boolean canContainDeletedFiles(ManifestFile manifest, boolean trustManifestReferences) {
+    if (manifest.minSequenceNumber() > 0 && manifest.minSequenceNumber() < minSequenceNumber) {
+      return true;
+    }
```

Comment:
I don't think we want to add this, I intentionally removed the behavior to rewrite a manifest in the case it only had aged out deletes, see here for a more detailed description https://github.com/apache/iceberg/pull/11131/files#r1815710103.

Reply:
If that's the case then doesn't the last condition here do nothing since it can only reach this check if one of the earlier checks was true anyway:

```java
 deletePaths.contains(file.location())
                          || deleteFiles.contains(file)
                          || dropPartitions.contains(file.specId(), file.partition())
                          || (isDelete
                              && entry.isLive()
                              && entry.dataSequenceNumber() > 0
                              && entry.dataSequenceNumber() < minSequenceNumber);
 ```
 
 It seems like perhaps `dropDeleteFilesOlderThan` has no affect anymore (unless `allDeletesReferenceManifests` gets set to false or something). 
 
I think not removing by `minSequenceNumber` leaves undeleted delete files that just never get applied to any files at query time, so it's not the end of the world, but it does lead to some wasted storage and slightly longer scan planning times. 

Assuming we want to keep this behaviour perhaps we should just not use `dropDeleteFilesOlderThan` anymore in `mergingSnapshotProducer` then?

Reply:
As a side note I think it's better to make sure `dropDeleteFilesOlderThan` does work. We have encountered tables that have many old EQ deleted files that were compacted and no longer apply to any data files, but were not removed from the table's manifests.

Reply:
Sorry for the delay, yes I believe you are right that the last condition specifically at https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/ManifestFilterManager.java#L427 is redundant, we can remove that! There is no longer any need to rely on that condition because the only time we'd hit that point is the manifest needs to be rewritten for correctness based on the other criteria. Then when the manifest does get rewritten if there happen to be aged out deletes, we will rewrite them here https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/ManifestFilterManager.java#L475

OK so breaking down the pros/cons for eagerly removing aged out deletes during commit where a manifest *only* has aged out deletes: 

The pro of eager removal of aged out deletes is that space consumed by metadata is reduced, and also planning times are at least slightly reduced. 

The downside of eager removal is that even if a manifest does not need to be rewritten, the commit process is incurring additional latency and risk of failure (the IO involved) for writing out a new manifest.

IMO I feel like the upside for reclaiming some metadata storage/reducing plan times, doesn't justify the risk of failure/additional latency being incurred by the process on the commit path. Maybe in the most extreme case where some large percentage of manifests _only_ have aged out deletes, is it worth it? 

I think you brought up good point though on rewrite manifests, the Java API itself will just retain all the delete manifests but I do think the spark action does support rewriting delete manifests https://github.com/apache/iceberg/blob/main/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java#L179

Perhaps it would be better for rewrite manifests to remove the aged out deletes? At that point the work for removing the aged out deletes is more intentional in the context of table maintenance and not on the general commit path for `MergingSnapshotProducer` implementations? I'll need to think more on this

Reply:
Perhaps it can be added as a table property. As an anecdote I have encountered tables with frequent updates that have 100K + inactive delete files in their manifests. And since Spark isn't used in this env performing a cleanup is simple.

I think the above issue should be addressed, but perhaps we got a bit side tracked in the context of this bug. How would you like to proceed here. This is still a valid bug fix, can you think of a trick to make the test reproduce the issue easily without reverting the change to  `dropDeleteFilesOlderThan`? If not can we merge it regardless, it's still a valid fix.